### PR TITLE
Adds privileged service to installer

### PR DESCRIPTION
**Pull request by Nino-K**
Signed-off-by: Nino Kodabande <nkodabande@suse.com>

### CODE REVIEW

File: build/elevated-install.ps1 [hunk 1]
```diff
@@ -119,10 +112,22 @@ function Install-Kernel {
...
+# Installs Rancher Desktop Privileged Service
+function Install-PrivilegedService {
+  Param([String] $InstallDir)
+  Write-Output "Installing Rancher Desktop Privileged Service"
+  $ExecPath = (Join-Path $InstallDir 'resources\win32\internal\privileged-service.exe')
+  $InstallCommand = "`"${InstallDir}`" install"
+  $Exec = Invoke-Expression -Command $InstallCommand
+  if ($Exec.ExitCode -ne '0') {
+    return 102
+  }
```
Comment (mook-as) [inline build/elevated-install.ps1:125]: >     PS C:\Users\Admin\rd\resources\win32\internal> .\privileged-service.exe install ; echo $LASTEXITCODE
>     Error: service [RancherDesktopPrivilegedService] already exists: The specified service already exists.
>     Usage:
>       privileged-service install [flags]
>     
>     Flags:
>       -h, --help   help for install
>     
>     1
We don't actually allow installing it twice — we probably should (and clobber the existing install?)

File: build/elevated-install.ps1 [hunk 2]
```diff
@@ -119,10 +112,22 @@ function Install-Kernel {
...
+# .SYNOPSIS
+# Installs Rancher Desktop Privileged Service
+function Install-PrivilegedService {
+  Param([String] $InstallDir)
+  Write-Output "Installing Rancher Desktop Privileged Service"
+  $ExecPath = (Join-Path $InstallDir 'resources\win32\internal\privileged-service.exe')
+  $InstallCommand = "`"${InstallDir}`" install"
+  $Exec = Invoke-Expression -Command $InstallCommand
+  if ($Exec.ExitCode -ne '0') {
+    return 102
```
Comment (mook-as) [inline build/elevated-install.ps1:124]: Should this be using `return` or `exit` here?

File: build/uninstall-privileged-service.ps1 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# Uninstalls Rancher Desktop Privileged Service
+function Uninstall-PrivilegedService {
+  Param([String] $InstallDir)
+  Write-Output "Uninstalling Rancher Desktop Privileged Service"
+  $InstallCommand = "`"${InstallDir}`" uninstall"
+  Invoke-Expression -Command $InstallCommand
+  exit $LASTEXITCODE
+}
+
+Uninstall-PrivilegedService (Join-Path $InstallDir 'resources\win32\internal\privileged-service.exe')
```
Comment (mook-as) [inline build/uninstall-privileged-service.ps1:16]: Doesn't this need to be run as administrator?

File: build/uninstall-privileged-service.ps1 [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+param($InstallDir)
+
+# .SYNOPSIS
+# Uninstalls Rancher Desktop Privileged Service
+function Uninstall-PrivilegedService {
+  Param([String] $InstallDir)
+  Write-Output "Uninstalling Rancher Desktop Privileged Service"
+  $InstallCommand = "`"${InstallDir}`" uninstall"
+  Invoke-Expression -Command $InstallCommand
```
Comment (mook-as) [inline build/uninstall-privileged-service.ps1:12]:
```suggestion
  & ${InstallDir} uninstall
```
Though presumably it should be called `$ServiceExecutable` or something along those lines instead.

**Review by mook-as — CHANGES_REQUESTED** [review]

File: build/elevated-install.ps1 (hunk 1 above)
Comment (Nino-K) [reply build/elevated-install.ps1:125]: Sure, I will make a separate PR for this one. We can merge that one first and this PR. WDYT?

File: build/uninstall-privileged-service.ps1 [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+param($ServiceExecutable)
+
+# .SYNOPSIS
+# Uninstalls Rancher Desktop Privileged Service
+function Uninstall-PrivilegedService {
+  Param([String] $ServiceExecutable)
+  Write-Output "Uninstalling Rancher Desktop Privileged Service"
+  $UninstallCommand = "`"${ServiceExecutable}`" uninstall"
+  $Process = (Start-Process -FilePath "${PSHOME}\PowerShell.exe" -Verb RunAs -Wait -PassThru -ArgumentList $UninstallCommand)
```
Comment (mook-as) [inline build/uninstall-privileged-service.ps1:12]: I'm not sure why we need to go through PowerShell here:
```suggestion
  $Process = (Start-Process -FilePath "${ServiceExecutable}" -Verb RunAs -Wait -PassThru -ArgumentList uninstall)
```

**Review by mook-as — APPROVED** [review]

**Review by mook-as — APPROVED** [review]

File: build/uninstall-privileged-service.ps1 [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+param($ServiceExecutable)
+
+# .SYNOPSIS
+# Uninstalls Rancher Desktop Privileged Service
+function Uninstall-PrivilegedService {
+  Param([String] $ServiceExecutable)
+  Write-Output "Uninstalling Rancher Desktop Privileged Service"
+  $UninstallCommand = "`"${ServiceExecutable}`" uninstall"
+  $Process = (Start-Process -FilePath "${ServiceExecutable}" -Verb RunAs -Wait -PassThru -ArgumentList $UninstallCommand)
```
Comment (mook-as) [inline build/uninstall-privileged-service.ps1:12]: You're passing the executable name as an argument to itself: `privileged-service.exe privleged-service.exe uninstall`

**Review by mook-as — CHANGES_REQUESTED** [review]

**Review by mook-as — APPROVED** [review]

**Pull request merged**